src/main/io/serial_cli.c

```diff
@@ -970,6 +973,61 @@ static void cliSerial(char *cmdline)
 
 }
 
+static void cliSerialPassthrough(char *cmdline)
+{
+    if (isEmpty(cmdline)) {
+        cliShowParseError();
+        return;
+    }
+
+    int id = -1;
+    unsigned baud = 0;
+    char* tok = strtok(cmdline, " ");
+    int index = 0;
+
+    while (tok != NULL) {
+        switch(index) {
+            case 0:
+                id = atoi(tok);
+                break;
+            case 1:
+                baud = atoi(tok);
+                break;
+        }
+        index++;
+        tok = strtok(NULL, " ");
+    }
+
+    serialPort_t *passThroughPort;
+    serialPortUsage_t *passThroughPortUsage = findSerialPortUsageByIdentifier(id);
+    if (!passThroughPortUsage || passThroughPortUsage->serialPort == NULL) {
+        if (!baud)
+        {
+            printf("Port %d is not open, you must specify baud\r\n", id);
+            return;
+        }
+
+        passThroughPort = openSerialPort(id, FUNCTION_PASSTHROUGH, NULL,
+                                         baudRates[baud], MODE_RXTX,
+                                         SERIAL_NOT_INVERTED);
+        if (!passThroughPort) {
+            printf("Port %d could not be opened\r\n", id);
+            return;
+        }
+        printf("Port %d opened, baud=%d\r\n", id, baudRates[baud]);
+    } else {
+        passThroughPort = passThroughPortUsage->serialPort;
+        if (!(passThroughPort->mode & MODE_TX))
```

Comment:
I can easily modify `serialPassThrough()` to "do the right thing" based on what mode the port is opened in by interrogating it's mode. Then it can become a one-way passthrough. Also can support an optional [mode] parameter to override this behavior.

I already fixed the double call to `serialSetMode()` and committed with a **duh** commit message. Just not pushed cleanup changes yet ;-)